src/firmware/app/control/wheel_controller.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include "firmware/shared/circular_buffer.h"
+
+typedef struct WheelController WheelController_t;
+
+/*
+ * NOTE: THIS FUNCTIONS IN THIS FILE OPERATE UNDER THE ASSUMPTION THAT THEY ARE BEING USED
+ * IN A DISCRETE TIME CONTROLLED SYSTEM. THIS MEANS THAT FOR A CONSTANT INTERPOLATION
+ * PERIOD OF 'T' ONLY ONE DATA SAMPLE IS ADDED TO THE SAMPLE BUFFER AND ONLY ONE COMMAND
+ * IS ADDED TO THE COMMAND BUFFER.
+ *
+ * THIS MEANS THAT OVER THE PERIOD OF 'T' SECONDS THERE IS NO CHANGE TO THE APPLIED
+ * VOLTAGE. THIS IS KNOWN AS 'ZERO-ORDER-HOLD' (ZOH).
+ *
+ * For more information:
+ * https://ethz.ch/content/dam/ethz/special-interest/mavt/dynamic-systems-n-control/idsc-dam/Lectures/Digital-Control-Systems/Slides_DigReg_2013.pdf
+ *
+ */
+
+/**
+ * Function creates a WheelController_t with the specified parameters and returns a
+ * pointer to it. The wheel_controller_destroy function MUST be called before the
+ * WheelController_t* goes out of scope.
+ * The WheelController_t allows previously sampled wheel states to be added to it's state,
+ * and allows for new wheel commands to be requested.
+ *
+ * The coefficients parameters are in order of decreasing recentness - where
+ * coefficient_array[0] is the most recent value.
+ *
+ * The convention is as follows:
+ * Where 'z' is the discrete time operator and z^M is equivalent to M time-steps BACKWARDS
+ * in time. coeff[0] -> z^0 coeff[1] -> z^-1 coeff[2] -> z^-2
+ * ...
+ * coeff[N] -> z^-N
+ *
+ * NOTE: All coefficients MUST be specified. That is if the difference equation is A*z^-3
+ * + B*z^-1 the z^-2 coefficient MUST be included. Ex, A*z^-3 + 0*z^-2 + B*z^-1.
+ *
+ * The difference equation created is:
+ *      Output_voltage = Command_coefficients*Command_values -
+ * Output_sample_coefficients*output_values
+ *
+ * @param command_coefficients [in] The coefficients for the controller command inputs.
```

Comment:
question: what does the `[in]` mean here? I thought it means integer but it is not

Reply:
The *[in* means that the parameter is an input. Apposed to [out] which means an output of the function

Reply:
got it